includes/ExternalRedirectInterceptor.php

```diff
@@ -0,0 +1,41 @@
+<?php
+namespace NewfoldLabs\WP\Module\Onboarding;
+
+use NewfoldLabs\WP\Module\Onboarding\Data\Data;
+
+/**
+ * Class to intercept redirect calls and filter them.
+ *
+ * The purpose of this class is to prevent any redirects while the user is on the onboarding page.
+ * The only allowed redirect is to the brand plugin page.
+ */
+class ExternalRedirectInterceptor {
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		if ( ! isset( $_GET['page'] ) || \sanitize_text_field( $_GET['page'] ) !== WP_Admin::$slug ) {
+			return;
+		}
+
+		\add_filter( 'wp_redirect', array( $this, 'wp_redirect' ), 10, 1 );
+	}
+
+	/**
+	 * Intercept wp_redirect calls and filter them.
+	 *
+	 * @param string $location The location to redirect to.
+	 */
+	public function wp_redirect( $location ): string {
+		$runtime_data = Data::runtime();
+		$brand_plugin_url  = $runtime_data['currentBrand']['pluginDashboardPage'];
```

Comment:
@wpalani What happens if one of these array keys doesn't exist?

Reply:
The runtime should always return a value for both `currentBrand` and `pluginDashboardPage`

https://github.com/newfold-labs/wp-module-onboarding-data/blob/main/includes/Data.php#L43-L49

Reply:
Yes, but given the brands defined: https://github.com/newfold-labs/wp-module-onboarding-data/blob/main/includes/Brands.php#L44-L471

And these HostGator locales: https://github.com/newfold-labs/wp-plugin-hostgator/blob/main/src/app/data/region.js#L9-L18

I'm wondering if it might still happen.

Reply:
If this may happen for smaller brands, we can be more defensive here since it's not a critical path feature (more of a firewall). I'll push a code to check the values gracefully before processing them.

Reply:
Here's the commit: https://github.com/newfold-labs/wp-module-onboarding/pull/632/commits/ca5e6533f31c23ef81fd81fb446a91ff036623b0